cmd/opentelemetry-collector/app/defaults/testdata/addprocessor-merged.yaml

```diff
@@ -0,0 +1,21 @@
+receivers:
+  jaeger:
+    protocols:
+      grpc:
+      thrift_compact:
+      thrift_binary:
+
+processors:
+  queued_retry: {}
+  batch:
+    timeout: 5s
+
+exporters:
+  jaeger_elasticsearch:
+
+service:
+  pipelines:
+    traces:
+      receivers: [jaeger]
+      processors: [queued_retry, batch]
```

Comment:
This is a good comment. We should allow overriding the order.

Reply:
In https://github.com/open-telemetry/opentelemetry-collector/pull/909` disabled flag has been removed from all components. Hence there is no way how a user could disable hardcoded/default component. 

In this PR I wanted to support incremental addition of components. For instance if a user wants to add a processor it would just add it to `services.pipelines.traces.processors` without repeating the default/hardcoded processors. Since there is no way to disable a component and order of the components might be important we should require defining an exact order of the components in the pipeline. 

Example: add attribute processor

```
processors:
  attributes:
    key: foo

service:
  pipelines:
    traces:
      processors: ["batch", "attributes"]
```
The `batch` has to be repeated otherwise it will not be included in the pipeline.

Reply:
Is this going to be considered/implemented in a separate PR?

Reply:
In this PR, as this will have to change anyways.